soccer/src/soccer/strategy/agent/position/defense.cpp

```diff
@@ -9,6 +10,10 @@ std::optional<RobotIntent> Defense::derived_get_task(RobotIntent intent) {
     return state_to_task(intent);
 }
 
+std::string Defense::return_current_state() {
+    return "Defense" + static_cast<int>(current_state_);
```

Comment:
Yeah, this doesn't work the way you think it does. Try this instead:

```suggestion
    return std::string{"Defense"} + std::to_string(static_cast<int>(current_state_));
```


The basic problem you have is that `"Offense"` is not of type `std::string` but rather `const char*`. It does get implicitly converted to an `std::string` but not before you add an integer to the pointer.